go/test/endtoend/transaction/twopc/fuzzer_test.go

```diff
@@ -0,0 +1,309 @@
+/*
+Copyright 2024 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package transaction
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"golang.org/x/exp/rand"
+
+	"vitess.io/vitess/go/mysql"
+)
+
+var (
+	// updateRowsVals are the rows that we use to ensure 1 update on each shard with the same increment.
+	updateRowsVals = [][]int{
+		{
+			4, // 4 maps to 0x20 and ends up in the first shard (-40)
+			6, // 6 maps to 0x60 and ends up in the second shard (40-80)
+			9, // 9 maps to 0x90 and ends up in the third shard (80-)
+			// We can increment all of these values by multiples of 16 and they'll always be in the same shard.
+		},
+	}
+
+	insertIntoFuzzUpdate   = "INSERT INTO twopc_fuzzer_update (id, col) VALUES (%d, %d)"
+	updateFuzzUpdate       = "UPDATE twopc_fuzzer_update SET col = col + %d WHERE id = %d"
+	insertIntoFuzzInsert   = "INSERT INTO twopc_fuzzer_insert (id, updateSet) VALUES (%d, '%s')"
+	selectFromFuzzUpdate   = "SELECT col FROM twopc_fuzzer_update WHERE id = %d"
+	selectIdFromFuzzInsert = "SELECT id FROM twopc_fuzzer_insert WHERE updateSet = '%s' ORDER BY col"
```

Comment:
`selectIdFromFuzzInsert` query used for validation is not clear.
I believe we can going to check the ordering of col to be in same order. 
May be I misunderstood what will be `id` value here.

Reply:
Yeah the id column was overloaded with the thread id information of the fuzzer thread that inserted it. Instead it is much simpler to reason and see the code if the fields are different. So I've split them up and this code has now changed.

Basically id is now only going to have 3 values, 4, 6 and 9 to ensure that the rows go into the 3 shards, and we'll store the thread id separately.

Then we can check that the threadIDs in the 3 shards should be the exact same for a given update set when ordered by the auto increment column, because that defines the ordering of commits on the shards.

Reply:
This holds till we do not start disrupting MySQL. Then these auto_increment across shards will get out of sync.

Reply:
We are only using col for ordering!